File: Sorting Plateau/Sorting/Q1. Minimum Absolute Difference/Q1_MinimumAbsoluteDifference.java
class Solution {
    // Solution class

    public List<List<Integer>> minimumAbsDifference(int[] arr) {
        // Method to find all pairs with minimum absolute difference

        List<List<Integer>> res = new ArrayList<>();
        // Stores the result pairs

        // In a sorted array, adjacent elements have the smallest difference
        Arrays.sort(arr);
        // Sort the array

        int min = Integer.MAX_VALUE;
        // Stores the minimum difference found so far

        for (int i = 0; i < arr.length - 1; i++) {
            // Loop through adjacent elements

            int diff = arr[i + 1] - arr[i];
            // Calculate difference between adjacent elements

            if (diff < min) {
                // Found a new smaller minimum difference

                min = diff;
                // Update minimum difference

                res.clear();
                // Clear old pairs

                res.add(Arrays.asList(arr[i], arr[i + 1]));
                // Add the new pair with minimum difference
            } 
            else if (diff == min) {
                // Found another pair with the same minimum difference

                res.add(Arrays.asList(arr[i], arr[i + 1]));
                // Add this pair to the result
            }
        }

        return res;
        // Return all pairs with minimum absolute difference
    }
}

File: Sorting Plateau/Sorting/Q1. Minimum Absolute Difference/Q1_MinimumAbsoluteDifference.py
class Solution:
    # Define Solution class

    def minimumAbsDifference(self, A):
        # Function to find all pairs with minimum absolute difference

        A.sort()
        # Sort the array in ascending order

        D = [A[i + 1] - A[i] for i in range(len(A) - 1)]
        # Store differences between adjacent elements

        target = min(D)
        # Find the minimum difference

        res = []
        # List to store result pairs

        for i, d in enumerate(D):
            # Loop through all differences with their index

            if d == target:
                # Check if difference equals minimum difference

                res.append([A[i], A[i + 1]])
                # Add the pair having minimum difference

        return res
        # Return all pairs with minimum absolute difference
